src/Network.cpp

```diff
@@ -267,6 +267,334 @@ std::pair<int, int> Network::load_v1_network(std::istream& wtfile) {
+        std::vector<float> weights = process(2);
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        m_conv_pol_b = std::move(weights);
+    }
+
+    // Size 2
+    {
+        std::vector<float> weights = process(m_bn_pol_w1.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_bn_pol_w1));
+    }
+
+    // Size 2
+    {
+        std::vector<float> weights = process(m_bn_pol_w2.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_bn_pol_w2));
+    }
+
+    // Size 261364
+    {
+        std::vector<float> weights = process(m_ip_pol_w.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_ip_pol_w));
+    }
+
+    // Size 362
+    {
+        std::vector<float> weights = process(m_ip_pol_b.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_ip_pol_b));
+    }
+
+    // Size filters
+    {
+        std::vector<float> weights = process(filters);
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        m_conv_val_w = std::move(weights);
+    }
+
+    // Size 2
+    {
+        std::vector<float> weights = process(2);
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        m_conv_val_b = std::move(weights);
+    }
+
+    // Size 1
+    {
+        std::vector<float> weights = process(m_bn_val_w1.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_bn_val_w1));
+    }
+
+    // Size 1
+    {
+        std::vector<float> weights = process(m_bn_val_w2.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_bn_val_w2));
+    }
+
+    // Size 92416
+    {
+        std::vector<float> weights = process(m_ip1_val_w.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_ip1_val_w));
+    }
+
+    // Size 256
+    {
+        std::vector<float> weights = process(m_ip1_val_b.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_ip1_val_b));
+    }
+
+    // Size 256
+    {
+        std::vector<float> weights = process(m_ip2_val_w.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_ip2_val_w));
+    }
+
+    // Size 1
+    {
+        std::vector<float> weights = process(m_ip2_val_b.size());
+        if (!weights.size()) {
+            return {0, 0};
+        }
+        std::copy(cbegin(weights), cend(weights), begin(m_ip2_val_b));
+    }
```

Comment:
This looks like a lot of repetitive code.  How about moving the code body into process() calls and handle the ```return {0,0};``` case as an exception throw?